internal/k8s/controller.go

```diff
@@ -837,6 +848,83 @@ func (lbc *LoadBalancerController) sync(task task) {
 	}
 }
 
+func (lbc *LoadBalancerController) updateWorkerProcessMetric() {
+	// get the current config version
+	client := nginx.NewVerifyClient(4000)
+	currentConfigVersion, err := client.GetConfigVersion()
+	if err != nil {
+		glog.Errorf("Error in getting the latest configuration version")
+	}
+
+	lsCmd := exec.Command("ls", "-l", "/proc")
+	grepCmd := exec.Command("grep", "-i", "nginx")
+	awkCmd := exec.Command("awk", "NF>1{print $NF}")
+
+	reader, writer := io.Pipe()
+	lsCmd.Stdout = writer
+	grepCmd.Stdin = reader
+
+	secReader, secWriter := io.Pipe()
+	grepCmd.Stdout = secWriter
+	awkCmd.Stdin = secReader
+
+	var buff bytes.Buffer
+	awkCmd.Stdout = &buff
+
+	lsCmd.Start()
```

Comment:
All these commands (Start, Wait, Close) can throw an error, not handling the errors is not a good idea. This means, this will end up in a huge method with a lot of `if err != nil`....

Apart form that, I find a little bit difficult to follow the fact that this is all in parts/lines. I am thinking of an approach where you can write all of this using Go code, for example, for the first part of getting PIDs, something like this:

```golang

        files, err := ioutil.ReadDir("/proc")
        if err != nil {
                fmt.Println(fmt.Sprintf("error %v", err))
        }

        for _, f := range files {
                u, _ := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
                if u.Name == "nginx user" {
                        // store pids of the processes
                        fmt.Println(u.Name)
                }
        }
```

And for the second part, follow the same approach of trying to find the Go code that does what you want instead of having to call shell 

What do you think?